framework/web/Application.php

```diff
@@ -55,7 +55,72 @@ class Application extends \yii\base\Application
      * @var Controller the currently active controller instance
      */
     public $controller;
+    /**
+     * Stack of middleware
+     * @var array
+     */
+    protected $middlewareStack = [];
```

Comment:
I'd probably prefer having a separate class to store middleware stack

Reply:
I thought about it. I wanted to make a collection, and MiddlewareItem class. But decided that this is memory. Should i make it?

Reply:
For my opinion yes, but it's a point to discuss. PHP 7.1 stores classes in memory as efficient as arrays, so it should not be an issue.